server/src/rest/routes/anonymousUsers.ts

```diff
@@ -0,0 +1,13 @@
+import { Router } from 'express';
+import { statusCodes } from '../../server';
+import { wrapResponse } from '../middleware';
+
+export const anonymousUsers = Router();
+
+anonymousUsers.post('/create', wrapResponse(
+  req => req.core.anonymousUsers.create(req.body),
+  statusCodes.STATUS_CREATED));
+
+anonymousUsers.post('/donate', wrapResponse(
+  req => req.core.anonymousUsers.donate(req.body)));
+
```

Comment:
I don't think it's a good idea to have 2 separate endpoints (i.e. one for creating anonymous users and one for donating anonymously). This `/donate` endpoint can be called with an arbitrary user's ID.

I think you should create only a single endpoint that should call a method like `users.donateAnonymously(args)`, the `args` should contain the amount as well as user details. This method should then call `users.createAnonymous()` and then `users.initiateDonation()`